src/main/java/databaseInteracting/DataBaseInteracting.java

```diff
@@ -0,0 +1,173 @@
+package databaseInteracting;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataBaseInteracting {
+  Connection connection;
+
+  /**
+   * Constructor of databaseInteracting class.
+   *
+   * @param url      String - url of the database connection.
+   * @param user     String - name of user to login on the database.
+   * @param password String - password to login on the database.
+   * @throws SQLException - Exception of database
+   */
+  public DataBaseInteracting(String url, String user, String password) throws SQLException {
+    DataBaseConnection dataBaseConnection =
+        new DatabaseInteracting.DataBaseConnection(url, user, password);
+    connection = dataBaseConnection.getConnection();
+  }
+
+  /**
+   * Method to prepare the query.
+   *
+   * @return preparedStatement of the query to list all products.
+   * @throws SQLException - Exception of database.
+   */
+  public PreparedStatement createPrepareStatementToProductList() throws SQLException {
+    String sql = "SELECT * FROM products";
+    return connection.prepareStatement(sql);
+  }
+
+  /**
+   * Method to add a product on the products table.
+   *
+   * @param name  String - name of the product.
+   * @param stock int - number of products on the stock.
+   * @param price float - price of the product.
+   * @throws SQLException - Exception of database.
+   */
+  public void addProductToDatabase(String name, int stock, float price) throws SQLException {
+    String sql = "INSERT INTO products (name, stock, price) VALUES (?,?,?)";
+    PreparedStatement preparedStatement = connection.prepareStatement(sql);
+    preparedStatement.setString(1, name);
+    preparedStatement.setInt(2, stock);
+    preparedStatement.setFloat(3, price);
+    preparedStatement.executeUpdate();
+    preparedStatement.close();
+  }
+
+  /**
+   * Method to get products by the id.
+   *
+   * @param id int - id of the product.
+   * @return a resultSet with the products.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet getProductsById(int id) throws SQLException {
+    PreparedStatement preparedStatement =
+        connection.prepareStatement("select * from products where id = ?");
+    preparedStatement.setInt(1, id);
```

Comment:
Prepared statement isn't being closed here

Reply:
Used a try with resources in the class that invoke this method to close the prepared statement.